### PR TITLE
Verify fvm contract

### PR DESCRIPTION
**Pull request by bitwiseguy**
This PR includes changes implemented in order to successfully verify a deployed contract on Filecoin Calibration testnet. This branch uses `hardhat` to deploy AND verify contracts, instead of trying to verify manually through a block explorer UI (which was a pain-staking, unsuccessful task).

The block explorer used for contract verification was [Starboard](https://fvm.starboard.ventures/calibration/explorer) because it was the only Filecoin Calibration explorer I could find with a readily accessible API and associated contract verification documentation. There is a nice etherscan hardhat plugin that can be used to verify against most common chains.

The `generate-nitro-adjudicator.sh` script was updated so that it uses the same `hardhat compile` command and settings as defined in `packages/nitro-protocol`. This prevents the repo from having two different smart contract compilation flows, which should help reduce the likelihood of errors when trying to verify contracts. This change closes https://github.com/statechannels/go-nitro/issues/911

This branch was used to deploy and verify a `NitroAdjudicator` contract on Calibration testnet: [link to contract](https://fvm.starboard.ventures/calibration/explorer/address/0xe1790Ea824035184a3Bf344e087FB61744992545). If we move the demo to that contract, this PR closes https://github.com/statechannels/go-nitro/issues/1767

### CODE REVIEW

**Review by lalexgap — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,5 +22,8 @@
     "lint": "yarn workspaces foreach -pt run lint",
     "test": "yarn workspaces foreach -pt run test"
   },
-  "packageManager": "yarn@3.5.1"
+  "packageManager": "yarn@3.5.1",
+  "dependencies": {
+    "ethers": "^6.7.1"
```
Comment (NiloCK) [inline package.json:27]: Curious: why was this added?

File: packages/nitro-protocol/hardhat-deploy/deploy-fvm.ts [hunk 2]
```diff
@@ -2,24 +2,22 @@ import 'hardhat-deploy';
 import 'hardhat-deploy-ethers';
 
 import {HardhatRuntimeEnvironment} from 'hardhat/types';
-import {BigNumber} from 'ethers';
```
Comment (NiloCK) [inline packages/nitro-protocol/hardhat-deploy/deploy-fvm.ts:5]: Same curiosity: was this import breaking the workflow for contract verification somehow? Or just a refactor to reduce the import of effectively duplicated type info?

**Review by NiloCK — APPROVED** [review]
Great :)

File: package.json (hunk 1 above)
Comment (bitwiseguy) [reply package.json]: Good catch. Removed in https://github.com/statechannels/go-nitro/pull/1789/commits/c92fcf19641294135ba12cb8930fef100e3f62b0. It was added by mistake

File: packages/nitro-protocol/hardhat-deploy/deploy-fvm.ts (hunk 2 above)
Comment (bitwiseguy) [reply packages/nitro-protocol/hardhat-deploy/deploy-fvm.ts:5]: This was not breaking the verification workflow, rather it is more of an optimization. Instead of using `ethers`, which is a rather large package, we are using the `hardhat-ethers` package which is already imported
